### PR TITLE
Do (more) client side validation of registration parameters.

### PR DESCRIPTION
**Pull request by dbkr**
Fixes https://github.com/vector-im/vector-web/issues/512

### CODE REVIEW

File: src/components/views/login/RegistrationForm.js [hunk 1]
```diff
@@ -50,31 +57,14 @@ module.exports = React.createClass({
...
-            errCode = "RegistrationForm.ERR_PASSWORD_MISMATCH";
-        }
-        else if (pwd1.length < this.props.minPasswordLength) {
-            errCode = "RegistrationForm.ERR_PASSWORD_LENGTH";
-        }
-        if (errCode) {
-            this.props.onError(errCode);
-            return;
-        }
-
```
Comment (Kegsay) [inline src/components/views/login/RegistrationForm.js]: We're removing the checks from "submit" time and moving them to be recalculated on the fly which is cool, but we then need to check in `onSubmit` that there are no invalid fields. I don't see this check being made, so what's stopping me from just hitting Register with invalid fields?

**Comment by Kegsay** [comment]
LGTM aside from comment.

File: src/components/views/login/RegistrationForm.js (hunk 1 above)
Comment (dbkr) [inline src/components/views/login/RegistrationForm.js]: The server will reject it anyway (don;t say we didn't warn you!) but yes, point - this is easy to do.

**Comment by dbkr** [comment]
Fixed. Changes are probably bug enough to warrant a ptal.

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**